File: listmenus/paiza4.swift
let n = Int(readLine()!)!
var dumbbellWeights: [Int] = []

for _ in 0..<n {
 let input = readLine()!.split(separator: " ")
    if input[0] == "ADD_LEFT" {
        dumbbellWeights.insert(Int(input[1])!, at: 0) 
    } 
    if input[0] == "ADD_RIGHT" {
        dumbbellWeights.append(Int(input[1])!)
    }
    if input[0] == "REMOVE_RIGHT" {
        dumbbellWeights.removeLast()
    }
    if input[0] == "REMOVE_LEFT" {
        dumbbellWeights.removeFirst()
    }
 }
 
 let weightSum = dumbbellWeights.reduce(0, +)
 print(weightSum)

File: listmenus/paiza4.py
Q = int(input())
dumbbell_weight_list = []

for i in range(Q):
  order = input().split()
  if order[0] == "ADD_RIGHT":
    weight = int(order[1])
    dumbbell_weight_list.append(weight)
  elif order[0] == "ADD_LEFT":
    weight = int(order[1])
    dumbbell_weight_list.insert(0, weight)
  elif order[0] == "REMOVE_RIGHT":
    dumbbell_weight_list.pop(-1)
  elif order[0] == "REMOVE_LEFT":
    dumbbell_weight_list.pop(0)

weight_sum = sum(dumbbell_weight_list)
print(weight_sum)